src/AppInstallerCLICore/Workflows/InstallFlow.cpp

```diff
@@ -364,11 +377,22 @@ namespace AppInstaller::CLI::Workflow
             }
             context << PortableInstall;
             break;
+        case InstallerTypeEnum::Zip:
+            context << ArchiveInstall;
+            break;
         default:
             THROW_HR(HRESULT_FROM_WIN32(ERROR_NOT_SUPPORTED));
         }
     }
 
+    void ArchiveInstall(Execution::Context& context)
+    {
+        context <<
+            ExtractInstallerFromArchive <<
+            VerifyAndSetNestedInstaller <<
+            ExecuteInstaller;

```

Comment:
This flow will be less efficient for portables; we should just extract directly to the final location.  Not sure if that works great with the database tracking files though.  We might need to be less efficient just so any files we could lose track of are in `%TEMP%` and will be more readily cleaned up.

Reply:
I am thinking that portables will require more tasks than the one I have currently specified. 
Here is what I am proposing when we start the implementation for portable(s) in an archive. 

- `ExtractInstallerFromArchive` (now called `ExtractFilesFromArchive`) will have a separate flow for portables that will copy the extracted files to the final install location
- `TryExtractArchive` will have an additional parameter to indicate that will append to the database to keep track of the files that are extracted and copied. 
- The install location will need to have the directory created prior to extraction, so this will likely need to be its own step that creates the InstallDirectory and the local db file if it does not exist already. So something like `SetupPortableInstallDirectory` that we call prior to all of these steps if the `NestedInstallerType` is `Portable`

Once we enter the portable install flow, we can then use `IsArchiveType` to check whether we should install multiple portables (I plan to use subcontexts to handle multiple installs) for a given archive package.